coap-core/src/main/java/com/mbed/coap/server/CoapServerBuilder.java

```diff
@@ -296,6 +312,11 @@ public CoapServerBuilderForUdp duplicateMsgDetectionTimeInMillis(long intervalIn
             return this;
         }
 
+        public CoapServerBuilderForUdp duplicateMessageDetectorCache(Cache<DuplicationDetector.CoapRequestId, CoapPacket> cache) {
```

Comment:
An internal class `DuplicationDetector` is exposed on the public interface.

Reply:
This is a bit tricky. I moved CoapRequestId out of duplicateDetector. Should the CoapRequestId be put public.
I would like to remove the timestamp from there first, because if it's public, it shouldn't be changed.

Reply:
@norbert-david  I think I would prefer to keep CoapRequestId private for now, kind of reserving the right to change it later.